main.go

```diff
@@ -85,20 +87,33 @@ func main() {
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "Warning: failed to determine zone: %s\n", err)
 			zone = ""
+			if *generateMeshId {
+				// Zone is required while using with generateMeshID.
+				os.Exit(1)
+			}
 		}
 	}
 
+	// Generate clusterName from metadata server or from command-line
+	// arguments, with the latter taking preference.
+	cluster := *gkeClusterName
+	if cluster == "" {
+		cluster, err = getClusterName()
```

Comment:
`err` is lost if `*generateMeshId == false`. It should be logged for `*includeDeploymentInfo` or something... but only if running on GKE...

I'd feel better if this block only ran if `*includeDeploymentInfo || *generateMeshId`. Although there's still the issue that it is dependent on the deployment type.

Reply:
Done. I made sure that within the *includeDeploymentInfo conditional logic, i print a warning if there was err from getting the cluter name from the metadata server